core/rawdb/freezer_remote_client.go

```diff
@@ -0,0 +1,112 @@
+package rawdb
+
+import (
+	"sync"
+
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+// FreezerRemoteClient is an RPC client implementing the interface of ethdb.AncientStore.
+// The struct's methods delegate the business logic to an external server
+// that is responsible for managing an actual ancient store.
+type FreezerRemoteClient struct {
+	client *rpc.Client
+	quit   chan struct{}
+	mu     sync.Mutex
+}
+
+const (
+	FreezerMethodClose            = "freezer_close"
+	FreezerMethodHasAncient       = "freezer_hasAncient"
+	FreezerMethodAncient          = "freezer_ancient"
+	FreezerMethodAncients         = "freezer_ancients"
+	FreezerMethodAncientSize      = "freezer_ancientSize"
+	FreezerMethodAppendAncient    = "freezer_appendAncient"
+	FreezerMethodTruncateAncients = "freezer_truncateAncients"
+	FreezerMethodSync             = "freezer_sync"
+)
+
+// newFreezerRemoteClient constructs a rpc client to connect to a remote freezer
+func newFreezerRemoteClient(endpoint string) (*FreezerRemoteClient, error) {
+	client, err := rpc.Dial(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	return &FreezerRemoteClient{
+		client: client,
+	}, nil
+}
+
+// Close terminates the chain freezer, unmapping all the data files.
+func (api *FreezerRemoteClient) Close() error {
+	api.mu.Lock()
```

Comment:
Is this mutex reasonable?

Reply:
Since the sibling FS `freezer` implementation _doesn't_ use a mutex around its methods, I think we can safely remove this. If freezer implementations want to implement a mutex at the server level, they can. Call atomicity should be managed by the caller and the implementer, not the middleman.

Reply:
Did with https://github.com/etclabscore/core-geth/pull/168/commits/3652204011544e7d04bf0ef317a7e5b1ae754a89. 

If you disagree, we'll revert.

Reply:
I approve of this